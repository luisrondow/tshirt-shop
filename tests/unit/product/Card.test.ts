import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import { createTestingPinia } from '@pinia/testing'

import { ProductCard } from '#components'

describe('ProductCard component', () => {
  it('renders product correctly', () => {
    const wrapper = mount(ProductCard, {
      global: {
        plugins: [createTestingPinia()],
      },
      props: {
        id: 'id-1',
        name: 'Product 1',
        price: 100,
        image: 'https://via.placeholder.com/150',
      },
    })

    expect(wrapper.text()).toContain('Product 1')
    expect(wrapper.text()).toContain('â‚¬100')
    expect(wrapper.find('img').attributes('src')).toBe('https://via.placeholder.com/150')
    expect(wrapper.find('button').text()).toBe('Add to basket')
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should emit add-to-basket event', async () => {
    const wrapper = mount(ProductCard, {
      global: {
        plugins: [createTestingPinia()],
      },
      props: {
        id: 'id-1',
        name: 'Product 1',
        price: 100,
        image: 'https://via.placeholder.com/150',
      },
    })

    await wrapper.find('button').trigger('click')

    expect(wrapper.emitted('add-to-basket')).toBeTruthy()
    expect(wrapper.emitted('add-to-basket')?.length).toBe(1)
  })

  it('should should shows View in basket when product is in basket', async () => {
    const wrapper = mount(ProductCard, {
      global: {
        stubs: {
          XLink: {
            template: '<a><slot /></a>',
          },
        },
        plugins: [
          createTestingPinia({
            initialState: { basket: { itemsInBasket: { 'id-1': 1 } } },
          }),
        ],
      },
      props: {
        id: 'id-1',
        name: 'Product 1',
        price: 100,
        image: 'https://via.placeholder.com/150',
      },
    })

    expect(wrapper.find('button').text()).toContain('View in Basket >')
  })
})
