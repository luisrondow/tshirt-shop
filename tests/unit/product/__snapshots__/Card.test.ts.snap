// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductCard component > renders product correctly 1`] = `
<div
  class="flex flex-col"
>
  <img
    alt="product"
    class="h-auto w-full object-cover"
    src="https://via.placeholder.com/150"
  />
  <div
    class="flex items-center justify-between py-4"
  >
    <h1
      class="font-serif text-2xl font-semibold"
    >
      Product 1
    </h1>
    <p
      class="font-mono text-xl font-semibold"
    >
      â‚¬100
    </p>
  </div>
  <button
    class="h-10 w-full uppercase border border-black bg-black px-4 py-2 text-white  hover:bg-white hover:text-black"
  >
    
     Add to basket 
    
  </button>
</div>
`;
